lib/workers/global/config/parse/file.ts

```diff
@@ -1,23 +1,47 @@
+import { readFile } from 'fs-extra';
+import { load } from 'js-yaml';
+import JSON5 from 'json5';
 import upath from 'upath';
 import { migrateConfig } from '../../../../config/migration';
-import type { AllConfig } from '../../../../config/types';
+import type { AllConfig, RenovateConfig } from '../../../../config/types';
 import { logger } from '../../../../logger';
 
+export async function getParsedContent(file: string): Promise<RenovateConfig> {
+  switch (upath.extname(file)) {
+    case '.yaml':
+    case '.yml':
+      return load(await readFile(file, 'utf8'), {
+        json: true,
+      }) as RenovateConfig;
+    case '.json5':
+    case '.json':
+      return JSON5.parse(await readFile(file, 'utf8'));
+    case '.js': {
+      const tmpConfig = await import(file);
+      return tmpConfig.default ? tmpConfig.default : tmpConfig;
+    }
+    default:
+      throw new Error('Unsupported file type');
+  }
+}
+
 export async function getConfig(env: NodeJS.ProcessEnv): Promise<AllConfig> {
-  let configFile = env.RENOVATE_CONFIG_FILE || 'config';
+  let configFile = env.RENOVATE_CONFIG_FILE || 'config.js';
```

Comment:
@rarkins This will break users who use `config.json` 🤔 This also breaks our helm chart

https://github.com/renovatebot/helm-charts/blob/51451d8e2f5e36b01af3972046f978d51c78bdad/charts/renovate/templates/cronjob.yaml#L69

Reply:
Should this PR be labeled `breaking` and go in a new major version then?

Reply:
I've added a breaking change trailer in one of my commits so that could be used as inspiration when you squash this PR 🐱 https://github.com/renovatebot/renovate/pull/12644#discussion_r757788590. Will push changes from the last review later today, thanks!

Reply:
I still think we should look for an existing `config.json` to not break at that file.

@rarkins @JamieMagee  WDYT?

Reply:
Let's look automatically for `config.js` and users need to specify the config file if it's `config.json`. It's a breaking change anyway.

Reply:
ok, created an issue
- https://github.com/renovatebot/helm-charts/issues/167

Reply:
Or is it only a small logic increase to check both files?

Reply:
@nejch can you work out if it's much effort to keep supporting both config.js and config.json automatically?

Reply:
@rarkins I believe I already had this here based on initial review (although it wasn't pretty) before we started this discussion of breaking it:
https://github.com/renovatebot/renovate/blob/42acf832e91312e98790da6be1ce8548104c6a9a/lib/workers/global/config/parse/file.ts#L30-L45

I can revert and see if I can clean it up a bit.

Reply:
leave as is, pr is breaking anyways, code looks too complicated